code/clue.js

```diff
@@ -6,54 +6,271 @@ const mrGreen = {
+  setTimeout(function() {
+    document.querySelector('.killer-deck').classList.remove('shakeX');
+    document.querySelector('.killer-deck').classList.remove('bounce');
+    document.querySelector('.killerIcon').classList.add('hide');
+
+    shuffleFavouriteWeapon();
+    mystery.killer = randomSelector(suspects)
+    document.getElementById('killerCard').style.background = mystery.killer.color
+    document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+    document.getElementById('killerAge').innerHTML = `${mystery.killer.age}`
+    document.getElementById('killerOccupation').innerHTML = `${mystery.killer.occupation}`
+    
+    document.getElementById('killerImage').src = `${mystery.killer.image}`
+    document.getElementById('killerDescription').innerHTML = `${mystery.killer.description}`
+
+    document.getElementById('favouriteWeapon').innerHTML = `Favourite weapon: ${mystery.killer.favouriteWeapon}`
+
+  }, 1500);
+}
+
+const pickWeapon = () => {
+  document.querySelector('.weapon-deck').classList.add('shakeX');
+  
+  setTimeout(function() {
+    document.querySelector('.weapon-deck').classList.add('bounce');
+  }, 1000)
+  
+  setTimeout(function() {
+    document.querySelector('.weapon-deck').classList.remove('bounce');
+    document.querySelector('.weapon-deck').classList.remove('shakeX');
+
+    const favouriteWeapon = weapons.find((item) => {
+        return item.id === mystery.killer.favouriteWeapon;
+      }
+    )
+    const weaponsClone = [
+      ...weapons,
+      favouriteWeapon,
+      favouriteWeapon,
+      favouriteWeapon,
+      favouriteWeapon,
+      favouriteWeapon,
+    ]
+    mystery.weapon = randomSelector(weaponsClone);
+  
+    document.getElementById('weaponName').innerHTML = `${mystery.weapon.name} of ${mystery.weapon.weight} pounds`
+  }, 1500);
 }
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+const pickRoom = () => {
+  document.querySelector('.room-deck').classList.add('shakeX');
+  
+  setTimeout(function() {
+    document.querySelector('.room-deck').classList.add('bounce');
+  }, 1000)
+  
+  setTimeout(function() {
+    document.querySelector('.room-deck').classList.remove('bounce');
+    document.querySelector('.room-deck').classList.remove('shakeX');
+
+    const suspectRooms = rooms.find((item) => {
+      return item === mystery.weapon.location;
+    });
+    const roomsClone = [
+      ...rooms,
+      suspectRooms,
+      suspectRooms,
+      suspectRooms,
+      suspectRooms,
+      suspectRooms,
+    ]
+    mystery.room = randomSelector(roomsClone);
 
-// STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
-// 'The murder was committed by Jacob Green, in the living room with a rope.'
+    document.getElementById('roomName').innerHTML = `${mystery.room}`
+  }, 1500);
+}
+
+const revealMystery = () => {
```

Comment:
Since I'm allowed to click the reveal crime button before I have clicked the cards, the return will be undefined. If you add an if / else statement here then you can return "no mystery has been solved" if no cards has been selected. And if they have been selected, you print out by who, whit what and in which room.